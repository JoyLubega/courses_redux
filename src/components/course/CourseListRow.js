import React from 'react';
import PropTypes from 'prop-types'
import {Link} from 'react-router-dom';

const CourseListRow = (props) => {
  const { course} = props;
  return (
    <tr>
      <td><a href={course.watchHref} target="_blank">Watch</a></td>
      <td><a href={'/course/' + course.id}>{course.title}</a></td>
      <td>{course.authorId}</td>
      <td>{course.category}</td>
      <td>{course.length}</td>
    </tr>
  );
};
CourseListRow.propTypes = {
  course: PropTypes.object.isRequired
};
export default CourseListRow;