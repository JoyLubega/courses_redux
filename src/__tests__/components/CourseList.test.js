import React from 'react';
import { shallow} from 'enzyme';
import CourseList from '../../components/course/CourseList';
import CourseListRow from '../../components/course/CourseListRow';


const setup=() =>{
    const props = {
      courses:[],
      
    }
    return shallow(<CourseList {...props} />);
}
test('renders the CourseList component', () => {
  const wrapper = setup(true);
  expect(wrapper.find('table')).toHaveLength(1);
});

describe('<CourseList /> component contains child componenets', () => {
  test('renders the CourseList component', () => {
    const wrapper = setup(true);
    expect(wrapper.find('CourseListRow')).toMatchSnapshot();
  });
});
