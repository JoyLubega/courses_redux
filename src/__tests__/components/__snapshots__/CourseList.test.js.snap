// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseList /> component contains child componenets renders the CourseList component 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): ReactWrapper {
    Symbol(enzyme.__unrendered__): <CourseList
      courses={Array []}
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "courses": Array [],
      },
      "ref": null,
      "rendered": Object {
        "instance": <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                 
              </th>
              <th>
                Title
              </th>
              <th>
                Author
              </th>
              <th>
                Category
              </th>
              <th>
                Length
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                   
                </th>
                <th>
                  Title
                </th>
                <th>
                  Author
                </th>
                <th>
                  Category
                </th>
                <th>
                  Length
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
          "className": "table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <thead>
              <tr>
                <th>
                   
                </th>
                <th>
                  Title
                </th>
                <th>
                  Author
                </th>
                <th>
                  Category
                </th>
                <th>
                  Length
                </th>
              </tr>
            </thead>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                   
                </th>
                <th>
                  Title
                </th>
                <th>
                  Author
                </th>
                <th>
                  Category
                </th>
                <th>
                  Length
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <tr>
                  <th>
                     
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Length
                  </th>
                </tr>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                       
                    </th>,
                    <th>
                      Title
                    </th>,
                    <th>
                      Author
                    </th>,
                    <th>
                      Category
                    </th>,
                    <th>
                      Length
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <th>
                       
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " ",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Title
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Title",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Title",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Author
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Author",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Author",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Category
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Category",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Category",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Length
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Length",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Length",
                    ],
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "thead",
          },
          Object {
            "instance": <tbody />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [
              Array [],
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "courses": Array [],
        },
        "ref": null,
        "rendered": Object {
          "instance": <table
            class="table"
          >
            <thead>
              <tr>
                <th>
                   
                </th>
                <th>
                  Title
                </th>
                <th>
                  Author
                </th>
                <th>
                  Category
                </th>
                <th>
                  Length
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                     
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Length
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
            "className": "table",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <thead>
                <tr>
                  <th>
                     
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Length
                  </th>
                </tr>
              </thead>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                     
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Length
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <tr>
                    <th>
                       
                    </th>
                    <th>
                      Title
                    </th>
                    <th>
                      Author
                    </th>
                    <th>
                      Category
                    </th>
                    <th>
                      Length
                    </th>
                  </tr>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <th>
                         
                      </th>,
                      <th>
                        Title
                      </th>,
                      <th>
                        Author
                      </th>,
                      <th>
                        Category
                      </th>,
                      <th>
                        Length
                      </th>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <th>
                         
                      </th>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": " ",
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                      ],
                      "type": "th",
                    },
                    Object {
                      "instance": <th>
                        Title
                      </th>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Title",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Title",
                      ],
                      "type": "th",
                    },
                    Object {
                      "instance": <th>
                        Author
                      </th>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Author",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Author",
                      ],
                      "type": "th",
                    },
                    Object {
                      "instance": <th>
                        Category
                      </th>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Category",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Category",
                      ],
                      "type": "th",
                    },
                    Object {
                      "instance": <th>
                        Length
                      </th>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Length",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Length",
                      ],
                      "type": "th",
                    },
                  ],
                  "type": "tr",
                },
              ],
              "type": "thead",
            },
            Object {
              "instance": <tbody />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [
                Array [],
              ],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "courses": Array [],
      },
      "ref": null,
      "rendered": Object {
        "instance": <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                 
              </th>
              <th>
                Title
              </th>
              <th>
                Author
              </th>
              <th>
                Category
              </th>
              <th>
                Length
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                   
                </th>
                <th>
                  Title
                </th>
                <th>
                  Author
                </th>
                <th>
                  Category
                </th>
                <th>
                  Length
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
          "className": "table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <thead>
              <tr>
                <th>
                   
                </th>
                <th>
                  Title
                </th>
                <th>
                  Author
                </th>
                <th>
                  Category
                </th>
                <th>
                  Length
                </th>
              </tr>
            </thead>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                   
                </th>
                <th>
                  Title
                </th>
                <th>
                  Author
                </th>
                <th>
                  Category
                </th>
                <th>
                  Length
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <tr>
                  <th>
                     
                  </th>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Category
                  </th>
                  <th>
                    Length
                  </th>
                </tr>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                       
                    </th>,
                    <th>
                      Title
                    </th>,
                    <th>
                      Author
                    </th>,
                    <th>
                      Category
                    </th>,
                    <th>
                      Length
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <th>
                       
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " ",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Title
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Title",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Title",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Author
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Author",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Author",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Category
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Category",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Category",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Length
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Length",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Length",
                    ],
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "thead",
          },
          Object {
            "instance": <tbody />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [
              Array [],
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
